src/scanspec/specs.py

```diff
@@ -1,3 +1,10 @@
+"""`Spec` and its subclasses.
```

Comment:
Nit: This is a self-referencing description, same for `regions.py` and `core.py`, it seems a bit redundant, similar to

```python
def get_foo() -> int:
    """
    gets foo
    """
    ...
```

Reply:
Ha, better words welcomed!